src/core/sys/posix/sys/ioccom.d

```diff
@@ -66,3 +66,64 @@ version (OSX)
         return _IOC!(T)(IOC_INOUT, cast(uint)g, cast(uint)n, T.sizeof);
     }
 }
+else version (FreeBSD)
+{
+    /* FreeBSD ioctl's encode the command in the lower 16-bits
+     * and the size of any in/out parameters in the lower 13 bits of the upper
+     * 16-bits of a 32 bit unsigned integer. The high 3 bits of the upper
+     * 16-bits encode the in/out status of the parameter.
+     */
+    enum uint IOCPARM_SHIFT = 13; // number of bits for ioctl size
+    enum uint IOCPARM_MASK = ((1 << IOCPARM_SHIFT) - 1); // parameter length mask
+    uint IOCPARM_LEN(uint x) // to extract the encoded parameter length
+    {
+        return ((x >> 16) & IOCPARM_MASK);
+    }
+    uint IOCBASECMD(uint x) // to extract the encoded command
+    {
+        return (x & ~(IOCPARM_MASK << 16));
+    }
+    uint IOCGROUP(uint x) // to extract the encoded group
+    {
+        return ((x >> 8) & 0xff);
+    }
+
+    enum uint IOCPARM_MAX = (1 << IOCPARM_SHIFT); // max size of ioctl args
+
+    enum uint IOC_VOID = 0x20000000; // no parameters
+    enum uint IOC_OUT = 0x40000000; // copy parameters back
+    enum uint IOC_IN = 0x80000000; // copy parameters into
+    enum uint IOC_INOUT = (IOC_IN | IOC_OUT);
+    enum uint IOC_DIRMASK = (IOC_VOID|IOC_OUT|IOC_IN);
+
+    // encode the ioctl info into 32 bits
+    uint _IOC(T=typeof(null))(uint inorout, uint group, uint num, size_t len)
+    {
+        return (inorout | ((len & IOCPARM_MASK) << 16) | (group << 8) | num);
+    }
+
+    // encode a command with no parameters
+    uint _IO(char g, int n)
+    {
+        return _IOC(IOC_VOID, cast(uint)g, cast(uint)n, cast(size_t)0);
+    }
+    uint _IOWINT(T=int)(char g, int n)
+    {
+        return _IOC(IOC_VOID, cast(uint)g, cast(uint)n, int.sizeof);
```

Comment:
T is unused, did you mean to use T.sizeof?

Reply:
Same as above. Removed not used template parameter.

Reply:
I had problems in the past with functions in the bindings not being available causing linker errors. This is most likely due to the bindings not being compiled. Can someone confirm this?  What I’m saying it that this might need to be a template

Reply:
This can be fixed later if it is an issue.